relay-server/src/actors/project_buffer.rs

```diff
@@ -265,18 +270,27 @@ impl BufferService {
         Ok(service)
     }
 
-    /// Checks the provided path and gets the size of the file this path points to.
-    fn estimate_buffer_size(path: Option<PathBuf>) -> Result<u64, BufferError> {
-        path.and_then(|path| std::fs::metadata(path).ok())
-            .map(|m| m.len())
-            .ok_or(BufferError::DatabaseFileError(Error::from(
-                ErrorKind::NotFound,
-            )))
+    /// Estimates the db size by multiplying `page_count * page_size`.
+    async fn estimate_buffer_size(db: &Pool<Sqlite>) -> Result<i64, BufferError> {
+        let mut rows = sqlx::query("pragma page_count; pragma page_size;").fetch(db);
+        let page_count: i64 = match rows.next().await {
+            Some(row) => row?.try_get(0)?,
+            None => return Err(BufferError::DatabaseError(sqlx::Error::RowNotFound)),
+        };
+        let page_size: i64 = match rows.next().await {
+            Some(row) => row?.try_get(0)?,
+            None => return Err(BufferError::DatabaseError(sqlx::Error::RowNotFound)),
+        };
+        let size = page_count * page_size;
+
+        relay_statsd::metric!(histogram(RelayHistograms::BufferDiskSize) = size as u64);
+
+        Ok(size)
```

Comment:
@olksdr I completely changed the way we compute the db size so we can log it as a metric more often. Let me know if you approve.

Reply:
Yeah, that would definitely work, and we do not have to all the time hit the filesystem.